pyomo/contrib/piecewise/tests/test_incremental.py

```diff
@@ -0,0 +1,251 @@
+        self.assertEqual(len(paraboloid_block.delta_one_constraint), 1)
+        self.assertIsInstance(paraboloid_block.x_constraint, Constraint)
+        self.assertEqual(len(paraboloid_block.x_constraint), 2)
+        self.assertIsInstance(paraboloid_block.set_substitute, Constraint)
+        self.assertEqual(len(paraboloid_block.set_substitute), 1)
+
+    # Test methods using the common_tests.py code.
+    def test_transformation_do_not_descend(self):
+        ct.check_transformation_do_not_descend(
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    def test_transformation_PiecewiseLinearFunction_targets(self):
+        ct.check_transformation_PiecewiseLinearFunction_targets(
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    def test_descend_into_expressions(self):
+        ct.check_descend_into_expressions(
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    def test_descend_into_expressions_constraint_target(self):
+        ct.check_descend_into_expressions_constraint_target(
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    def test_descend_into_expressions_objective_target(self):
+        ct.check_descend_into_expressions_objective_target(
+            self, 'contrib.piecewise.incremental', make_log_x_model_ordered()
+        )
+
+    @unittest.skipUnless(SolverFactory('gurobi').available(), 'Gurobi is not available')
+    @unittest.skipUnless(SolverFactory('gurobi').license_is_valid(), 'No license')
+    def test_solve_log_model(self):
+        m = make_log_x_model_ordered()
+        TransformationFactory('contrib.piecewise.incremental').apply_to(m)
+        TransformationFactory('gdp.bigm').apply_to(m)
+        SolverFactory('gurobi').solve(m)
+        ct.check_log_x_model_soln(self, m)
+
+    # Failed during development when j1 vertex ordering got broken
+    @unittest.skipUnless(SolverFactory('gurobi').available(), 'Gurobi is not available')
+    @unittest.skipUnless(SolverFactory('gurobi').license_is_valid(), 'No license')
+    def test_solve_product_model(self):
+        m = ConcreteModel()
+        m.x1 = Var(bounds=(0.5, 5))
+        m.x2 = Var(bounds=(0.9, 0.95))
+        pts = list(itertools.product([0.5, 2.75, 5], [0.9, 0.925, 0.95]))
+        m.pwlf = PiecewiseLinearFunction(
+            points=pts,
+            function=lambda x, y: x * y,
+            triangulation=Triangulation.OrderedJ1,
+        )
+        m.obj = Objective(sense=minimize, expr=m.pwlf(m.x1, m.x2))
+        TransformationFactory("contrib.piecewise.incremental").apply_to(m)
+        SolverFactory('gurobi').solve(m)
+        self.assertAlmostEqual(0.45, value(m.obj))
+
+
+# Make a version of the log_x model with the simplices properly ordered for the
+# incremental transform
+def make_log_x_model_ordered():
```

Comment:
Instead of copying the model, could you perhaps just modify the current one to have a callback for building the piecewise part and then you just have to reimplement that for the ordered case? (Sorry to be picky, but have been burned before by having a million copies of every test and test model for a bunch of different almost-the-same cases.)